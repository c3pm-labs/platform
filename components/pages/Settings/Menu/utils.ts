enum Options {
  PROFILE = 'Profile',
  SECURITY = 'Security',
}

export default Options;
