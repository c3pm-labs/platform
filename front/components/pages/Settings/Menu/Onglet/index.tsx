import { useState, useEffect } from 'react';
import {
  makeStyles, Typography,
} from '@material-ui/core';
import clsx from 'clsx';

import Options from '../utils';

const useStyles = makeStyles((theme) => ({
  container: {
    transition: '0.5s',
    height: '2.5em',
    minHeight: '2.5em',
    maxHeight: '2.5em',
    flex: 1,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    width: '70%',
    border: 'none',
    backgroundColor: 'rgba(0,0,0,0)',
    padding: 0,
    '&:hover': {
      opacity: 0.8,
      cursor: 'pointer',
    },
    '&:focus': {
      outline: 0,
    },
  },
  greyOptions: {
    color: `${theme.palette.text.primary}`,
  },
  containerVerticalLine: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    Width: '4px',
    minWidth: '4px',
    maxWidth: '4px',
  },
  verticalLine: {
    borderRadius: `${theme.shape.borderRadius}px`,
    display: 'flex',
    height: '2.5em',
    border: 'solid',
    width: '0px',
  },
  verticalLineFocused: {
    borderWidth: '2px',
    borderColor: `${theme.palette.primary.main}`,
  },
  verticalLineBlur: {
    borderWidth: '0.5px',
    borderColor: `${theme.palette.text.disabled}`,
  },
  containerTile: {
    display: 'flex',
    justifyContent: 'flex-start',
    alignItems: 'center',
    Width: '90px',
    minWidth: '90px',
    maxWidth: '90px',
  },
}));

interface OngletsProps {
  currentFocus: Options;
  title: Options;
  setIsMenu: (newValue: boolean) => void;
  onClick: (newValue: Options) => void;
}

function Onglet(props: OngletsProps): JSX.Element {
  const {
    currentFocus,
    title,
    onClick,
    setIsMenu,
  } = props;
  const classes = useStyles();
  const [isFocus, setIsFocus] = useState(false);

  useEffect(() => {
    if (currentFocus === title) setIsFocus(true);
    else setIsFocus(false);
  }, [currentFocus, title]);

  const handleClassNameVerticalLine = () => (
    clsx((isFocus ? classes.verticalLineFocused : classes.verticalLineBlur), classes.verticalLine)
  );

  return (
    <button type="button" className={classes.container} onClick={(): void => { onClick(title); setIsMenu(false); }}>
      <div className={classes.containerVerticalLine}>
        <div className={handleClassNameVerticalLine()} />
      </div>
      <div className={classes.containerTile}>
        <Typography className={classes.greyOptions}>
          {title}
        </Typography>
      </div>
    </button>
  );
}

export default Onglet;
