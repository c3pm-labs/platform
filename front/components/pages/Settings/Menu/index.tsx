import React from 'react';
import {
  makeStyles, Typography,
} from '@material-ui/core';
import { useTranslation } from 'next-i18next';

import Onglet from './Onglet';
import Options from './utils';

const useStyles = makeStyles((theme) => ({
  container: {
    height: '140px',
    width: '160px',
    maxWidth: '160px',
    display: 'flex',
    flex: 1,
    borderWidth: '1px',
    border: 'solid',
    borderRadius: `${theme.shape.borderRadius}px`,
    borderColor: `${theme.palette.text.disabled}`,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    marginLeft: theme.spacing(2),
    [theme.breakpoints.down('xs')]: {
      marginLeft: 0,
      fontSize: '1.2em',
      maxWidth: '200px',
    },
  },
  blueTitle: {
    color: `${theme.palette.primary.main}`,
    fontWeight: 500,
    fontSize: '1.2em',
    paddingBottom: `${theme.spacing(1)}px`,
  },
  options: {
    display: 'flex',
    width: '100%',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
}));

interface MenuProps {
  currentFocus: Options;
  setCurrentFocus: (newValue: Options) => void;
  setIsMenu: (newValue: boolean) => void;
}

function Menu(props: MenuProps): JSX.Element {
  const {
    currentFocus,
    setCurrentFocus,
    setIsMenu,
  } = props;
  const classes = useStyles();
  const { t } = useTranslation('common');

  return (
    <div className={classes.container}>
      <Typography className={classes.blueTitle}>
        {t('settings.menu.title')}
      </Typography>
      <div className={classes.options}>
        <Onglet
          title={Options.PROFILE}
          currentFocus={currentFocus}
          onClick={setCurrentFocus}
          setIsMenu={setIsMenu}
        />
        <Onglet
          title={Options.SECURITY}
          currentFocus={currentFocus}
          onClick={setCurrentFocus}
          setIsMenu={setIsMenu}
        />
      </div>
    </div>
  );
}

export default Menu;
