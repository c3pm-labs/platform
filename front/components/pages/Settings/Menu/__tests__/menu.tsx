import { render } from '@testing-library/react';

import Menu from '../index';
import Options from '../utils';

test('Menu', () => {
  const { container } = render(
    <Menu
      currentFocus={'Profile' as Options}
      // eslint-disable-next-line no-console
      setIsMenu={(value) => console.log('set this menu', value)}
      // eslint-disable-next-line no-console
      setCurrentFocus={(value) => console.log('click', value)}
    />,
  );

  expect(container).toMatchSnapshot();
});
