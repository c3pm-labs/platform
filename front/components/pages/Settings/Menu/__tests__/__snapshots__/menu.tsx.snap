// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu 1`] = `
<div>
  <div
    class="makeStyles-container-1"
  >
    <p
      class="MuiTypography-root makeStyles-blueTitle-2 MuiTypography-body1"
    >
      Settings :
    </p>
    <div
      class="makeStyles-options-3"
    >
      <button
        class="makeStyles-container-4"
        type="button"
      >
        <div
          class="makeStyles-containerVerticalLine-6"
        >
          <div
            class="makeStyles-verticalLineFocused-8 makeStyles-verticalLine-7"
          />
        </div>
        <div
          class="makeStyles-containerTile-10"
        >
          <p
            class="MuiTypography-root makeStyles-greyOptions-5 MuiTypography-body1"
          >
            Profile
          </p>
        </div>
      </button>
      <button
        class="makeStyles-container-4"
        type="button"
      >
        <div
          class="makeStyles-containerVerticalLine-6"
        >
          <div
            class="makeStyles-verticalLineBlur-9 makeStyles-verticalLine-7"
          />
        </div>
        <div
          class="makeStyles-containerTile-10"
        >
          <p
            class="MuiTypography-root makeStyles-greyOptions-5 MuiTypography-body1"
          >
            Security
          </p>
        </div>
      </button>
    </div>
  </div>
</div>
`;
